Client/Client.Qt/Settings.cpp

```diff
@@ -19,17 +19,29 @@ Settings& Settings::getInstance()
 void Settings::writeSettings()
 {
     settings->beginGroup("Font");
-    settings->setValue("FontSize", _fontSize);
+    settings->setValue("ChatFontSize", _fontSize);
     settings->endGroup();
 }
 
-void Settings::setFontSize(int size)
+void Settings::resetSettings()
 {
-    _fontSize = size;
+    settings->clear();
 }
 
-int Settings::getFontSize()
+void Settings::setFontSize(std::uint32_t size)
 {
-    return _fontSize;
+    if (size > _minFontSize) 
+        _fontSize = size;
+    else 
+        throw std::range_error("font size value must be >0");
```

Comment:
just ignore if 0

Reply:
@StasDash I see it differently since after calling the method, it's expected to execute some stuff. If no exception is thrown and the argument doesn't suit the condition, wouldn't it seem weird that some method was called and nothing happened?
Also there is a recommendation on our wiki to use exceptions:
![image](https://user-images.githubusercontent.com/55056330/129281419-88243f73-b5e7-449a-aab8-7cf54dcdb0b8.png)
Could you please clarify why in this particular situation exception isn't a good practice?

Reply:
> @StasDash I see it differently since after calling the method, it's expected to execute some stuff. If no exception is thrown and the argument doesn't suit the condition, wouldn't it seem weird that some method was called and nothing happened?

No.

> Also there is a recommendation on our wiki to use exceptions:
> ![image](https://user-images.githubusercontent.com/55056330/129281419-88243f73-b5e7-449a-aab8-7cf54dcdb0b8.png)
> Could you please clarify why in this particular situation exception isn't a good practice?

If you throw exception application will just crash since we there is no catcher. And also if you set it font size to zero(or less) qt will just set it to system default size.

Reply:
@StasDash If we rely on Qt, why was the input check so necessary?

Reply:
we don't rely on qt, there should be no red message